### ISSUE TITLE
The performance for 'Refined seed for AMN (CAM + CRF)'

### ISSUE THREAD

**Issue by Zhengyang1995** [gaviotas/AMN#3]
Hey, thank you for your impressive work.

I try to download your 'Refined seed for AMN (CAM + CRF)', it is the pseudo mask for train_aug dataset.
I don't see if you published the evaluation function for train_aug, so I used the 'eval_sem_seg.py', it could see test the performance for 'train' and 'val' set.
But the miou of 'train' set is only among 37.78%.

I also used 'evaluation.py' in https://github.com/YudeWang/SEAM to get the miou of 'train_aug', the result is among 33.23%.

I am not sure if I made any mistakes during the evaluation, just let me know if I did and thank you again for your work.

**Comment by gaviotas** [comment]
Hello, @Zhengyang1995 

The refined seed (CAM + CRF) includes an undefined region (the value is 255).

However, the undefined region in the prediction is not ignored in both 'eval_sem_seg.py' and 'evaluation.py'.
(The undefined region of the ground-truth mask is only ignored.)

Therefore, all pixels in the undefined region in the prediction are considered wrong predictions during evaluation, leading to lower performance.

If you only want to evaluate valid pixels in the prediction, you should ignore them during the evaluation.
(e.g., 'evaluation.py' in SEAM, you can change the line 37, `cal = gt<255` ==> `cal = (gt<255 & prediction<255)` )

Please check it and let me know if you have any further questions :)

**Comment by Zhengyang1995** [comment]
Thank you! After changed, the performance is evaluated as expected.
Since I don't find the description of 'ignore' in your paper, would you please let me know the way you define 'undefined' region and how do you use it in the following steps?
1. Do you use the same way as 'IRNet' to get the undefined region, likes the class activations lower than a threshold?
2. In the following steps (likes taking refined seed as pseudo mask), do you ignore them, or take them as background?

Thank you so much for your explanation. I really appreciate that you could make such promising performance just by CAM+dCRF. It is so impressive.

**Comment by gaviotas** [comment]
The answers to the two questions are as follows:

1. For CAM+dCRF, we use the same procedure as IRN, which utilizes the class activations below the fg threshold and above the bg threshold. https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/step/cam_to_ir_label.py#L27-L35

2. The refined seed (CAM + dCRF) is used as the target for training AMN. During training AMN, undefined pixels are ignored when calculating the loss function.
https://github.com/gaviotas/AMN/blob/7f612d00e434eab7766ec517473cdf1b9d639192/step/train_amn.py#L40

After training AMN, we generate the final pseudo-masks using the refinement technique, IRN. In the final pseudo-masks, there is no undefined region. Finally, we train a segmentation network with the generated pseudo-masks as supervision.

If you have further questions, please leave an issue :)

Thanks.

**Comment by Zhengyang1995** [comment]
Thank you so much for your explanation！
